compiler/src/dmd/cparse.d

```diff
@@ -5987,6 +5989,92 @@ final class CParser(AST) : Parser!AST
+                            //printf("functionlike %s\n", id.toChars());
+
+                            // Capture the parameter list
+                            VarArg varargs = VarArg.none;
+                            auto parameters = new AST.Parameters();
+                            nextToken();        // skip past `(`
+                        Lwhile:
+                            while (1)
+                            {
+                                if (token.value == TOK.rightParenthesis)
+                                    break;
+                                if (token.value == TOK.dotDotDot)
+                                {
+                                    static if (0)       // variadic macros not supported yet
+                                    {
+                                        varargs = AST.VarArg.variadic;  // C-style variadics
+                                        nextToken();
+                                        if (token.value == TOK.rightParenthesis)
+                                            break Lwhile;
+                                    }
+                                    break Lswitch;
+                                }
+
+                                if (token.value != TOK.identifier)
+                                    break Lswitch;
+                                auto param = new AST.Parameter(token.loc, 0, null, token.ident, null, null);
+                                parameters.push(param);
+                                nextToken();
+                                if (token.value == TOK.comma)
+                                {
+                                    nextToken();
+                                    continue;
+                                }
+                                break;
+                            }
+                            if (token.value != TOK.rightParenthesis)
+                                break;
+                            nextToken();
+                            auto parameterList = AST.ParameterList(parameters, varargs, 0);
+                            /* Create a type for each parameter. Add it to the template parameter list,
+                             * and the parameter list.
+                             */
+                            auto tpl = new AST.TemplateParameters();
+                            foreach (param; (*parameters)[])
+                            {
+                                auto idtype = Identifier.generateId("__MP");
+                                auto loc = param.loc;
+                                auto tp = new AST.TemplateTypeParameter(loc, idtype, null, null);
+                                tpl.push(tp);
+
+                                auto at = new AST.TypeIdentifier(loc, idtype);
+                                param.type = at;
+                            }
+
+                            eLatch.sawErrors = false;
+                            auto exp = cparseExpression();
```

Comment:
I suspect this is the problematic line.

`#define`s are constrained to a single line, but `cparseExpression` ignores whitespace, so if there's a problem parsing, then it consumes too much.

```
#define parse_error()
#define eaten_by_cparseExpression 0x1
int foo = eaten_by_cparseExpression; // undefined 
```